tests/parsers/methodical_mind/methodical_mind_structure_test.py

```diff
@@ -1,73 +1,92 @@
-            contents=test_file, original_file_name="original"
+def test_create_header() -> None:
+    data = SeriesData(
+        pd.Series(
+            {
+                "FileName": "Z:\\MSD-WP\\Export\\22M0TALA85_2023-09-08-133933.txt",
+                "Version": "MMPR 1.0.38",
+                "Model": "HTS",
+                "Serial #": "1298873245",
+            }
         )
-        file_lines = read_to_lines(named_file_contents)
-        reader = CsvReader(file_lines)
-        combined_data = CombinedData.create(reader)
-
-    assert (
-        combined_data.file_name
-        == "Z:\\MSD-WP\\Export\\22M0TALA85_2023-09-08-133933.txt"
     )
-    assert combined_data.version == "MMPR 1.0.38"
-    assert combined_data.model == "HTS"
-    assert combined_data.serial_number == "1298873245"
-    assert combined_data.plate_doc_info[0].plate_well_count == 96
-    assert len(combined_data.plate_doc_info) == 2
+    assert Header.create(data) == Header(
+        file_name="Z:\\MSD-WP\\Export\\22M0TALA85_2023-09-08-133933.txt",
+        version="MMPR 1.0.38",
+        model="HTS",
+        serial_number="1298873245",
+    )
 
 
-def test_create_get_parameter() -> None:
-    lines = [
-        "FileName :\tZ:\\DC\\Export\\test_file.txt",
-        "Version  :\tMMPR 1.0.38",
-        "",
-        "User     :\tJeffy P",
+def test_create_plate_well_data_multiple_values_per_well() -> None:
+    header = SeriesData(
+        pd.Series(
+            {
+                "Read Time": "2022-01-01",
+                "User": "Jeffy P",
+                "Barcode1": "WP002",
+            }
+        )
+    )
+    data = [
+        [8213, 3475, 5589, 2131, 761],
+        [8062, 2341, 3147, 1846, 67778],
+        [463, 1485, 15855, 9587, 3226],
+        [2181, 6665, 70504, 40572, 13664],
+        [1345, 4737, 80463, 34564, 12378],
+        [839, 2692, 247, 128, 5024],
     ]
-    reader = CsvReader(lines)
-    assert CombinedData.get_parameter(reader, "User") == "Jeffy P"
+    plate_df = pd.DataFrame(
+        # The repeated label row labels signify multiple rows per well
+        data,
+        index=["A", "A", "A", "B", "B", "B"],
+        columns=["1", "2", "3", "4", "5"],
+    )
 
+    result = PlateData.create(header, plate_df)
 
-def test_create_get_parameter_when_missing() -> None:
-    lines = [
-        "FileName :\tZ:\\DC\\Export\\test_file.txt",
-        "Version  :\tMMPR 1.0.38",
-        "",
-        "User     :",
-    ]
-    reader = CsvReader(lines)
-    assert CombinedData.get_parameter(reader, "User") is None
+    assert result.plate_well_count == 10
+    assert len(result.well_data) == 30
+    assert result.well_data[0].luminescence == 8213
+    assert result.well_data[0].location_identifier == "A1_1"
+    assert result.well_data[-1].luminescence == 5024
+    assert result.well_data[-1].location_identifier == "B5_3"
 
 
-def test_create_plate_well_data() -> None:
-    data = {
-        "Unnamed: 0": ["A", np.nan, np.nan, np.nan, np.nan],
```

Comment:
Note to reviewer - the dataframe input changed here because we do the data fixing in the Reader class now, to `PlateData.create` expects cleaned up data.